identity/src/main/java/com/android/identity/util/SimpleApplicationData.java

```diff
@@ -0,0 +1,252 @@
+        } else {
+            mApplicationData.put(key, value.getBytes(StandardCharsets.UTF_8));
+        }
+
+        try {
+            mSaveFunc.call();
+        } catch (Exception e) {
+            Logger.d("SimpleApplicationData", "Unexpected exception occurred: " + e);
+        }
+    }
+
+    @Override
+    public void setApplicationData(@NonNull String key, @Nullable Long value) {
+        if (value == null) {
+            mApplicationData.remove(key);
+        } else {
+            mApplicationData.put(key, Util.cborEncodeNumber(value));
+        }
+
+        try {
+            mSaveFunc.call();
+        } catch (Exception e) {
+            Logger.d("SimpleApplicationData", "Unexpected exception occurred: " + e);
+        }
+    }
+
+    @Override
+    public void setApplicationData(@NonNull String key, @Nullable Timestamp value) {
+        if (value == null) {
+            mApplicationData.remove(key);
+        } else {
+            mApplicationData.put(key, Util.cborEncodeNumber(value.toEpochMilli()));
+        }
+
+        try {
+            mSaveFunc.call();
+        } catch (Exception e) {
+            Logger.d("SimpleApplicationData", "Unexpected exception occurred: " + e);
+        }
+    }
+
+    @Override
+    public void setApplicationData(@NonNull String key, @Nullable Boolean value) {
+        if (value == null) {
+            mApplicationData.remove(key);
+        } else {
+            mApplicationData.put(key, Util.cborEncodeBoolean(value));
+        }
+
+        try {
+            mSaveFunc.call();
+        } catch (Exception e) {
+            Logger.d("SimpleApplicationData", "Unexpected exception occurred: " + e);
+        }
+    }
+
+    @Nullable
+    @Override
+    public byte[] getApplicationData(@NonNull String key) {
+        return mApplicationData.get(key);
+    }
+
+    @Nullable
+    @Override
+    public String getApplicationDataString(@NonNull String key) {
+        byte[] value = mApplicationData.get(key);
+        if (value == null) {
+            return null;
+        }
+        return new String(value, StandardCharsets.UTF_8);
+    }
+
+    @Nullable
+    @Override
+    public Long getApplicationDataNumber(@NonNull String key) {
+        byte[] value = mApplicationData.get(key);
+        if (value == null) {
+            return null;
+        }
+        return Util.cborDecodeLong(value);
+    }
+
+    @Nullable
+    @Override
+    public Timestamp getApplicationDataTimestamp(@NonNull String key) {
+        byte[] value = mApplicationData.get(key);
+        if (value == null) {
+            return null;
+        }
+        return Timestamp.ofEpochMilli(Util.cborDecodeLong(value));
+    }
+
+    @Nullable
+    @Override
+    public Boolean getApplicationDataBoolean(@NonNull String key) {
+        byte[] value = mApplicationData.get(key);
+        if (value == null) {
+            return null;
+        }
+        return Util.cborDecodeBoolean(value);
+    }
+
+    @Override
+    public byte[] serialize() {
```

Comment:
Instead of `serialize()` and `deserialize()` I would use the names `encodeAsCbor()` and `fromEncodedCbor()` since it's slightly more descriptive.